From 27f990fb55f83bfb33c5d6e42c982daa30d327f7 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Thu, 15 Sep 2011 19:22:12 +0800
Subject: ENGR00156849 MX6Q: add relative clock for BCH

The BCH needs the pl301_mx6qperl_bch clock.
The BCH will not work if the clock is not enabled.
So add it.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 arch/arm/mach-mx6/clock.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-mx6/clock.c b/arch/arm/mach-mx6/clock.c
index 3f2ac85..dbb7dc8 100644
--- a/arch/arm/mach-mx6/clock.c
+++ b/arch/arm/mach-mx6/clock.c
@@ -3869,11 +3869,20 @@ static struct clk gpmi_nfc_clk[] = {
 	{	/* bch_apb_clk */
 	__INIT_CLK_DEBUG(gpmi_bch_apb_clk)
 	.parent = &usdhc3_clk,
+	.secondary = &gpmi_nfc_clk[4],
 	.enable = _clk_enable,
 	.enable_reg = MXC_CCM_CCGR4,
 	.enable_shift = MXC_CCM_CCGRx_CG12_OFFSET,
 	.disable = _clk_disable,
 	},
+	{	/* bch relative clk */
+	__INIT_CLK_DEBUG(pl301_mx6qperl_bch)
+	.parent = &usdhc3_clk,
+	.enable = _clk_enable,
+	.enable_reg = MXC_CCM_CCGR4,
+	.enable_shift = MXC_CCM_CCGRx_CG6_OFFSET,
+	.disable = _clk_disable,
+	},
 };
 
 static struct clk pwm_clk[] = {
@@ -4112,6 +4121,7 @@ static struct clk_lookup lookups[] = {
 	_REGISTER_CLOCK(NULL, "gpmi-apb", gpmi_nfc_clk[1]),
 	_REGISTER_CLOCK(NULL, "bch", gpmi_nfc_clk[2]),
 	_REGISTER_CLOCK(NULL, "bch-apb", gpmi_nfc_clk[3]),
+	_REGISTER_CLOCK(NULL, "pl301_mx6qperl-bch", gpmi_nfc_clk[4]),
 	_REGISTER_CLOCK("mxc_pwm.0", NULL, pwm_clk[0]),
 	_REGISTER_CLOCK("mxc_pwm.1", NULL, pwm_clk[1]),
 	_REGISTER_CLOCK("mxc_pwm.2", NULL, pwm_clk[2]),
-- 
1.7.9.5

